R/fixed_design_ahr.R

```diff
@@ -132,6 +132,9 @@ fixed_design_ahr <- function(
     input = input, enroll_rate = d$enroll_rate,
     fail_rate = d$fail_rate, analysis = ans, design = "ahr"
   )
+  attr(y, "design_display") <- "Average hazard ratio"
+  attr(y, "title") <- "Fixed Design under AHR Method"
+  attr(y, "footnote") <- "Power computed with average hazard ratio method."
```

Comment:
This can be done in a single `structure()` call, e.g.,

```suggestion
  y <- structure(
    y, design_display = "Average hazard ratio",
    title = "Fixed Design under AHR Method",
    footnote = "Power computed with average hazard ratio method."
  )
```

Or even include everything in the `structure()` call:

```r
  y <- structure(
    list(
      input = input, enroll_rate = d$enroll_rate,
      fail_rate = d$fail_rate, analysis = ans, design = "ahr"
    ),
    class = "fixed_design",
    design_display = "Average hazard ratio",
    title = "Fixed Design under AHR Method",
    footnote = "Power computed with average hazard ratio method."
  )
```

Feel free to pick a form that you are comfortable with.

Reply:
Nice suggestion. I hadn't even considered using `structure()`. I agree this cleans everything up. Done in https://github.com/Merck/gsDesign2/pull/497/commits/1b2e5fde2bd2aad19ad655ff8137e8692a3c3200

The only complication was that some of the other `fixed_design_X()` footnotes depend on `design_display`, so for these I defined `design_display` prior to the call to `structure()`.